lib/setup.js

```diff
@@ -249,18 +253,22 @@ function processCommand(command, args, params, callback) {
             if (args[0] === 'custom') {
                 setup.setupCustom(callback);
             } else {
-                let i = 0;
                 let isFirst;
                 let isRedis;
-                while (args[i] !== undefined) {
-                    if (args[i] === 'first' || args[i] === '--first') {
+
+                // we support "first" and "redis" without "--" flag
+                for (const arg of args) {
+                    if (arg === 'first') {
                         isFirst = true;
-                    } else if (args[i] === 'redis' || args[i] === '--redis') {
```

Comment:
of course, --flags will be in params and not in args, so `--redis` and `--first` were non-working, however in general `setup first` worked in combination with `redis` flag as desired in #687. Furthermore, I've added the functionality to specify the port. Specifying the host has already been possible via `--objects ip` and `--states ip`. Maybe we should also distinguish between ports for states and objects if we allow different hosts?